src/vivarium_gates_child_iv_iron/components/maternal_interventions.py

```diff
@@ -66,83 +76,117 @@ def setup(self, builder: Builder) -> None:
         self.bep_exposure_pipeline = self._get_exposure_pipeline(
             builder,
             self.bep_exposure_pipeline_name,
-            self._get_bep_exposure
+            self._get_bep_exposure,
+            [self.supplementation_exposure_column_name],
         )
         self.ifa_exposure_pipeline = self._get_exposure_pipeline(
             builder,
             self.ifa_exposure_pipeline_name,
-            self._get_ifa_exposure
+            self._get_ifa_exposure,
+            [self.supplementation_exposure_column_name],
         )
         self.mmn_exposure_pipeline = self._get_exposure_pipeline(
             builder,
             self.mmn_exposure_pipeline_name,
-            self._get_mmn_exposure
+            self._get_mmn_exposure,
+            [self.supplementation_exposure_column_name],
         )
+        self.iv_iron_exposure_pipeline = self._get_exposure_pipeline(
+            builder,
+            self.iv_iron_exposure_pipeline_name,
+            self._get_iv_iron_exposure,
+            [self.iv_iron_exposure_column_name],
+        )
+
         self.population_view = self._get_population_view(builder)
 
         self._register_simulant_initializer(builder)
 
     def _register_simulant_initializer(self, builder: Builder) -> None:
         builder.population.initializes_simulants(
             self.on_initialize_simulants,
-            creates_columns=[self.exposure_column_name],
+            creates_columns=[
+                self.supplementation_exposure_column_name,
+                self.iv_iron_exposure_column_name,
+            ],
         )
 
-    def _get_exposure_pipeline(self, builder: Builder, supplementation: str, exposure_source: Callable) -> Pipeline:
+    @staticmethod
+    def _get_exposure_pipeline(
```

Comment:
I do not understand why you want wrappers around the builder interfaces.  It seems like pointless indirection.  These are model-specific components.  We should prefer simplicity where we can.

Reply:
Fair point. It's not necessary or particularly useful in this case.